core/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -351,40 +352,24 @@ class DeltaCatalog extends DelegatingCatalogExtension
 
     override def commitStagedChanges(): Unit = {
       val conf = spark.sessionState.conf
-      val props = new util.HashMap[String, String]()
-      // Options passed in through the SQL API will show up both with an "option." prefix and
-      // without in Spark 3.1, so we need to remove those from the properties
-      val optionsThroughProperties = properties.asScala.collect {
-        case (k, _) if k.startsWith("option.") => k.stripPrefix("option.")
-      }.toSet
-      val sqlWriteOptions = new util.HashMap[String, String]()
-      properties.asScala.foreach { case (k, v) =>
-        if (!k.startsWith("option.") && !optionsThroughProperties.contains(k)) {
-          // Do not add to properties
-          props.put(k, v)
-        } else if (optionsThroughProperties.contains(k)) {
-          sqlWriteOptions.put(k, v)
-        }
-      }
-      if (writeOptions.isEmpty && !sqlWriteOptions.isEmpty) {
-        writeOptions = sqlWriteOptions.asScala.toMap
+      val catalogProp = DeltaTableProperties(properties)
+      if (writeOptions.isEmpty && catalogProp.options.nonEmpty) {
+        writeOptions = catalogProp.options
       }
-      if (conf.getConf(DeltaSQLConf.DELTA_LEGACY_STORE_WRITER_OPTIONS_AS_PROPS)) {
+      val props = if (conf.getConf(DeltaSQLConf.DELTA_LEGACY_STORE_WRITER_OPTIONS_AS_PROPS)) {
         // Legacy behavior
-        writeOptions.foreach { case (k, v) => props.put(k, v) }
+        catalogProp.properties ++ writeOptions
       } else {
-        writeOptions.foreach { case (k, v) =>
-          // Continue putting in Delta prefixed options to avoid breaking workloads
-          if (k.toLowerCase(Locale.ROOT).startsWith("delta.")) {
-            props.put(k, v)
-          }
-        }
+        val deltaPrefix: Map[String, String] = writeOptions.filter { case (k, _) =>
+          k.toLowerCase(Locale.ROOT).startsWith("delta.") }
+
+        catalogProp.properties ++ deltaPrefix
```

Comment:
This should also be added to `DeltaTableProperties`. We would like to be consistent in all places.